### PR TITLE
Fix missing key warning

### PR DESCRIPTION
**Pull request by Janpot**
I added some testing infra that catches these warnings as well. It also caught a warning on the `Home` for `Text nodes cannot appear as a child of <tbody>.` which I also fix in this PR. cc @bharatkashyap

### CODE REVIEW

**Comment by apedroferreira** [comment]
> Should we add a check for certain warnings in the integration tests?

Shouldn't eslint catch that kind of stuff if we add the proper rules?

**Comment by Janpot** [comment]
> Shouldn't eslint catch that kind of stuff if we add the proper rules?

It seems to miss this specific case. It is enabled for the file though, I can trigger it using another pattern (`const x = [<div />, <div />]`).

An integration test could catch these and possible more React warnings

**Comment by apedroferreira** [comment]
I've never used integration tests for code related issues but if you think it's a valid approach I would be curious to see how it would work :) I guess it could be useful to catch the issues that eslint can't.

**Comment by Janpot** [comment]
> I would be curious to see how it would work

We can capture console messages in playwright and throw when we detect certain patterns

**Comment by apedroferreira** [comment]
> > I would be curious to see how it would work
> 
> We can capture console messages in playwright and throw when we detect certain patterns

ok that's what i was thinking, sounds good!

File: packages/toolpad-app/src/toolpad/Home.tsx [hunk 1]
```diff
@@ -463,33 +464,32 @@ function AppsListView({
...
-                        onDelete={() => setDeletedApp(app)}
-                      />
-                    );
-                  })
-                : 'No apps yet';
-            default:
-              return '';
-          }
-        })()}
+        {status === 'loading' ? (
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/Home.tsx:478]: just nitpicking but maybe it's slightly easier to understand if this loading check is done above like the error check, and then here we use the ternary operator to check if there are no apps or more?

File: test/integration/smoke.spec.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import { test, expect, Request } from '@playwright/test';
+import { test, expect, Request } from '../playwright/test';
 import generateId from '../utils/generateId';
 import * as locators from '../utils/locators';
 
-test('basic app creation flow', async ({ page }) => {
+test.only('basic app creation flow', async ({ page }) => {
```
Comment (apedroferreira) [inline test/integration/smoke.spec.ts:5]: we should have an eslint rule for these if we don't, i missed it before in my PRs too

File: packages/toolpad-app/src/toolpad/Home.tsx (hunk 1 above)
Comment (Janpot) [reply packages/toolpad-app/src/toolpad/Home.tsx]: I refactored it a bit for readability

File: test/integration/smoke.spec.ts (hunk 2 above)
Comment (Janpot) [reply test/integration/smoke.spec.ts]: I added the `--forbid-only` flag in CI

**Review by apedroferreira — APPROVED** [review]

**Pull request merged**